File: komputer-sdk/typescript/src/client.test.ts
import { describe, it, expect } from "vitest";
import { KomputerClient } from "./client";

describe("KomputerClient", () => {
  it("instantiates with default URL", () => {
    const client = new KomputerClient();
    expect(client).toBeDefined();
  });

  it("instantiates with custom URL", () => {
    const client = new KomputerClient("http://example.com:8080");
    expect(client).toBeDefined();
  });

  it("strips trailing slash from URL", () => {
    const client = new KomputerClient("http://localhost:8080/");
    expect(client).toBeDefined();
  });
});

describe("Agent methods exist", () => {
  const client = new KomputerClient();

  it("has createAgent", () => expect(typeof client.createAgent).toBe("function"));
  it("has listAgents", () => expect(typeof client.listAgents).toBe("function"));
  it("has getAgent", () => expect(typeof client.getAgent).toBe("function"));
  it("has patchAgent", () => expect(typeof client.patchAgent).toBe("function"));
  it("has deleteAgent", () => expect(typeof client.deleteAgent).toBe("function"));
  it("has cancelAgentTask", () => expect(typeof client.cancelAgentTask).toBe("function"));
  it("has getAgentEvents", () => expect(typeof client.getAgentEvents).toBe("function"));
});

describe("Memory methods exist", () => {
  const client = new KomputerClient();

  it("has createMemory", () => expect(typeof client.createMemory).toBe("function"));
  it("has listMemories", () => expect(typeof client.listMemories).toBe("function"));
  it("has getMemory", () => expect(typeof client.getMemory).toBe("function"));
  it("has patchMemory", () => expect(typeof client.patchMemory).toBe("function"));
  it("has deleteMemory", () => expect(typeof client.deleteMemory).toBe("function"));
});

describe("Skill methods exist", () => {
  const client = new KomputerClient();

  it("has createSkill", () => expect(typeof client.createSkill).toBe("function"));
  it("has listSkills", () => expect(typeof client.listSkills).toBe("function"));
  it("has getSkill", () => expect(typeof client.getSkill).toBe("function"));
  it("has patchSkill", () => expect(typeof client.patchSkill).toBe("function"));
  it("has deleteSkill", () => expect(typeof client.deleteSkill).toBe("function"));
});

describe("Schedule methods exist", () => {
  const client = new KomputerClient();

  it("has createSchedule", () => expect(typeof client.createSchedule).toBe("function"));
  it("has listSchedules", () => expect(typeof client.listSchedules).toBe("function"));
  it("has getSchedule", () => expect(typeof client.getSchedule).toBe("function"));
  it("has patchSchedule", () => expect(typeof client.patchSchedule).toBe("function"));
  it("has deleteSchedule", () => expect(typeof client.deleteSchedule).toBe("function"));
});

describe("Secret methods exist", () => {
  const client = new KomputerClient();

  it("has createSecret", () => expect(typeof client.createSecret).toBe("function"));
  it("has listSecrets", () => expect(typeof client.listSecrets).toBe("function"));
  it("has updateSecret", () => expect(typeof client.updateSecret).toBe("function"));
  it("has deleteSecret", () => expect(typeof client.deleteSecret).toBe("function"));
});

describe("Connector methods exist", () => {
  const client = new KomputerClient();

  it("has createConnector", () => expect(typeof client.createConnector).toBe("function"));
  it("has listConnectors", () => expect(typeof client.listConnectors).toBe("function"));
  it("has getConnector", () => expect(typeof client.getConnector).toBe("function"));
  it("has deleteConnector", () => expect(typeof client.deleteConnector).toBe("function"));
});

describe("Office methods exist", () => {
  const client = new KomputerClient();

  it("has listOffices", () => expect(typeof client.listOffices).toBe("function"));
  it("has getOffice", () => expect(typeof client.getOffice).toBe("function"));
  it("has deleteOffice", () => expect(typeof client.deleteOffice).toBe("function"));
});

File: komputer-sdk/go/client_test.go
package sdk

import (
	"testing"
)

func TestNew(t *testing.T) {
	c := New("http://localhost:8080")
	if c == nil {
		t.Fatal("New() returned nil")
	}
	if c.api == nil {
		t.Fatal("client.api is nil")
	}
}

func TestNewTrailingSlash(t *testing.T) {
	c := New("http://localhost:8080/")
	if c == nil {
		t.Fatal("New() returned nil")
	}
}

// TestMethodsExist verifies all convenience methods compile and are addressable.
func TestMethodsExist(t *testing.T) {
	c := New("http://localhost:8080")

	methods := []interface{}{
		// Agents
		c.CreateAgent,
		c.ListAgents,
		c.GetAgent,
		c.DeleteAgent,
		c.PatchAgent,
		c.CancelAgentTask,
		c.GetAgentEvents,
		c.WatchAgent,
		// Memories
		c.CreateMemory,
		c.ListMemories,
		c.GetMemory,
		c.PatchMemory,
		c.DeleteMemory,
		// Skills
		c.CreateSkill,
		c.ListSkills,
		c.GetSkill,
		c.PatchSkill,
		c.DeleteSkill,
		// Schedules
		c.CreateSchedule,
		c.ListSchedules,
		c.GetSchedule,
		c.PatchSchedule,
		c.DeleteSchedule,
		// Secrets
		c.CreateSecret,
		c.ListSecrets,
		c.UpdateSecret,
		c.DeleteSecret,
		// Connectors
		c.CreateConnector,
		c.ListConnectors,
		c.GetConnector,
		c.DeleteConnector,
		c.ListConnectorTools,
		// Offices
		c.ListOffices,
		c.GetOffice,
		c.DeleteOffice,
		c.GetOfficeEvents,
	}

	for i, m := range methods {
		if m == nil {
			t.Errorf("method %d is nil", i)
		}
	}
}

func TestCreateAgentOpts(t *testing.T) {
	opts := CreateAgentOpts{
		Model:     ptrString("claude-sonnet-4-6"),
		Lifecycle: ptrString("Sleep"),
		Skills:    []string{"skill-1"},
		Memories:  []string{"mem-1"},
	}
	if *opts.Model != "claude-sonnet-4-6" {
		t.Errorf("expected claude-sonnet-4-6, got %s", *opts.Model)
	}
	if *opts.Lifecycle != "Sleep" {
		t.Errorf("expected Sleep, got %s", *opts.Lifecycle)
	}
	if len(opts.Skills) != 1 || opts.Skills[0] != "skill-1" {
		t.Errorf("expected [skill-1], got %v", opts.Skills)
	}
}

func ptrString(s string) *string { return &s }
